src/api/app/controllers/trigger_workflow_controller.rb

```diff
@@ -14,12 +14,19 @@ class TriggerWorkflowController < ApplicationController
 
   before_action :set_token
   before_action :validate_token_type
-  before_action :set_workflow_run
+
+  # overwrite the exception handling inherited from the rescue handler
+  rescue_from ActiveRecord::RecordInvalid do |exception|
+    if exception.record.errors[:hook_action].any? || exception.record.errors[:hook_event].any?
+      render_ok(data: { info: exception.record.errors.full_messages.to_sentence })
+    else
+      render_error(status: 400, message: exception.record.errors.full_messages.to_sentence)
+    end
+  end
 
   def create
     authorize @token, :trigger?
-
-    @workflow_run.save!
+    create_workflow_run
```

Comment:
Those are just two lines, no need to stuff them into a method...

Reply:
Moved the content of `create_workflow_run` to `create`.